## Data Cleaning and Dashboard cration

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("tobacco_data.csv")
df

Sl. No States & Union Territories Women (15 years & above)  \
0      1.0         Andaman &  Nicobar                   31.3 %   
1      2.0                         AP                     3.8%   
2      3.0                  Arunachal                    18.8%   
3      4.0                      Assam                    22.1%   
4      5.0                      Bihar                     5.0%   
5      6.0                 Chandigarh                     0.6%   
6      7.0               Chattishgarh                    17.3%   
7      8.0                        D&N                     2.9%   
8      9.0                      Delhi                     2.2%   
9     10.0                        Goa                     2.6%   
10    11.0                    Gujarat                     8.7%   
11    12.0                    Haryana                     2.5%   
12    13.0           Himachal Pradesh                     1.7%   
13    14.0                        J&K                     3.6%   
14    15.0                  Jharkhand                     8.4%   
15    16.0                  Karnataka                     8.5%   
16    17.0                     Kerala                     2.2%   
17    18.0                     Ladakh                     3.2%   
18    19.0                Lakshadweep                    17.5%   
19    20.0                Maharashtra                    10.9%   
20    21.0                    Manipur                    43.1%   
21    22.0                  Meghalaya                    28.2%   
22    23.0                    Mizoram                    61.6%   
23    24.0             Madhya Pradesh                    10.2%   
24    25.0                   Nagaland                    13.7%   
25    26.0                     Odisha                    26.0%   
26    27.0                 Puducherry                     2.6%   
27    28.0                     Punjab                     0.4%   
28    29.0                  Rajasthan                     6.9%   
29    30.0                     Sikkim                    11.7%   
30    31.0                 Tamil Nadu                     4.9%   
31    32.0                  Telangana                     5.6%   
32    33.0                    Tripura                    50.4%   
33    34.0              Uttar Pradesh                     8.4%   
34    35.0                Uttarakhand                     4.6%   
35    36.0                West Bengal                    10.8%   
36     NaN                      India                     8.9%   

   Men (15 years & above)  
0                  58.7 %  
1                   22.6%  
2                   50.3%  
3                   51.8%  
4                   48.8%  
5                   12.1%  
6                   43.1%  
7                   38.6%  
8                   26.3%  
9                   18.2%  
10                  41.1%  
11                  29.1%  
12                  32.3%  
13                  38.3%  
14                  47.4%  
15                  27.1%  
16                  16.9%  
17                  35.7%  
18                  28.5%  
19                  33.8%  
20                  58.1%  
21                  57.7%  
22                  72.9%  
23                  46.5%  
24                  48.4%  
25                  51.6%  
26                  14.8%  
27                  12.9%  
28                  42.0%  
29                  41.3%  
30                  20.1%  
31                  22.3%  
32                  56.9%  
33                  44.1%  
34                  33.7%  
35                  48.1%  
36                  38.0%

In [3]:
df = df.drop(36)

In [4]:
df

Sl. No States & Union Territories Women (15 years & above)  \
0      1.0         Andaman &  Nicobar                   31.3 %   
1      2.0                         AP                     3.8%   
2      3.0                  Arunachal                    18.8%   
3      4.0                      Assam                    22.1%   
4      5.0                      Bihar                     5.0%   
5      6.0                 Chandigarh                     0.6%   
6      7.0               Chattishgarh                    17.3%   
7      8.0                        D&N                     2.9%   
8      9.0                      Delhi                     2.2%   
9     10.0                        Goa                     2.6%   
10    11.0                    Gujarat                     8.7%   
11    12.0                    Haryana                     2.5%   
12    13.0           Himachal Pradesh                     1.7%   
13    14.0                        J&K                     3.6%   
14    15.0                  Jharkhand                     8.4%   
15    16.0                  Karnataka                     8.5%   
16    17.0                     Kerala                     2.2%   
17    18.0                     Ladakh                     3.2%   
18    19.0                Lakshadweep                    17.5%   
19    20.0                Maharashtra                    10.9%   
20    21.0                    Manipur                    43.1%   
21    22.0                  Meghalaya                    28.2%   
22    23.0                    Mizoram                    61.6%   
23    24.0             Madhya Pradesh                    10.2%   
24    25.0                   Nagaland                    13.7%   
25    26.0                     Odisha                    26.0%   
26    27.0                 Puducherry                     2.6%   
27    28.0                     Punjab                     0.4%   
28    29.0                  Rajasthan                     6.9%   
29    30.0                     Sikkim                    11.7%   
30    31.0                 Tamil Nadu                     4.9%   
31    32.0                  Telangana                     5.6%   
32    33.0                    Tripura                    50.4%   
33    34.0              Uttar Pradesh                     8.4%   
34    35.0                Uttarakhand                     4.6%   
35    36.0                West Bengal                    10.8%   

   Men (15 years & above)  
0                  58.7 %  
1                   22.6%  
2                   50.3%  
3                   51.8%  
4                   48.8%  
5                   12.1%  
6                   43.1%  
7                   38.6%  
8                   26.3%  
9                   18.2%  
10                  41.1%  
11                  29.1%  
12                  32.3%  
13                  38.3%  
14                  47.4%  
15                  27.1%  
16                  16.9%  
17                  35.7%  
18                  28.5%  
19                  33.8%  
20                  58.1%  
21                  57.7%  
22                  72.9%  
23                  46.5%  
24                  48.4%  
25                  51.6%  
26                  14.8%  
27                  12.9%  
28                  42.0%  
29                  41.3%  
30                  20.1%  
31                  22.3%  
32                  56.9%  
33                  44.1%  
34                  33.7%  
35                  48.1%

In [5]:
import pandas as pd
import dash
import dash_core_components as dcc
import dash_html_components as html
import plotly.express as px


C:\Users\ARNAB\AppData\Local\Temp\ipykernel_9292\1617444197.py:3: UserWarning: 
The dash_core_components package is deprecated. Please replace
`import dash_core_components as dcc` with `from dash import dcc`
  import dash_core_components as dcc
C:\Users\ARNAB\AppData\Local\Temp\ipykernel_9292\1617444197.py:4: UserWarning: 
The dash_html_components package is deprecated. Please replace
`import dash_html_components as html` with `from dash import html`
  import dash_html_components as html


In [6]:
# Create a Dash app
app = dash.Dash(__name__)

# Define the layout of the dashboard
app.layout = html.Div([
    # Dropdown to select between Women and Men
    dcc.Dropdown(
        id='gender-dropdown',
        options=[
            {'label': 'Women (15 years & above)', 'value': 'Women (15 years & above)'},
            {'label': 'Men (15 years & above)', 'value': 'Men (15 years & above)'}
        ],
        value='Women (15 years & above)',
        style={'width': '50%'}
    ),
    # Graph to display the state-wise comparison
    dcc.Graph(id='gender-graph')
])

# Define a callback to update the graph based on the selected gender
@app.callback(
    dash.dependencies.Output('gender-graph', 'figure'),
    [dash.dependencies.Input('gender-dropdown', 'value')]
)
def update_graph(selected_gender):
    fig = px.bar(df, x='States & Union Territories', y=selected_gender,
                 labels={'States & Union Territories': 'States'},
                 title=f'State-wise Comparison of {selected_gender}',
                 color_discrete_sequence=['pink' if selected_gender == 'Women (15 years & above)' else 'blue']
                 )
    return fig


if __name__ == '__main__':
    app.run_server(debug=True)


Based on the data provided, we can draw some conclusions about tobacco use among adults in different states and union territories in India. The table provides the percentage of women and men (aged 15 years and above) who use tobacco in each region:

1. The data suggests significant variations in tobacco use prevalence across different states and union territories in India.
2. Mizoram has the highest tobacco use prevalence among women, with 61.6%, followed by Manipur with 43.1%.
3. Among men, Mizoram also has the highest tobacco use prevalence, with 72.9%, followed closely by Andaman & Nicobar with 58.7%.
4. Chandigarh has the lowest tobacco use prevalence among women and men, with 0.6% and 12.1%, respectively.
5. There are several states and union territories where the percentage of men using tobacco is higher than the percentage of women using tobacco, such as Andaman & Nicobar, Arunachal Pradesh, Assam, Bihar, and Manipur.
6. However, in Mizoram, the percentage of women using tobacco is higher than the percentage of men using tobacco, indicating a unique trend in that region.
7. The data highlights the need for targeted interventions and awareness campaigns to address the regional disparities and promote tobacco control efforts in areas with higher prevalence rates.

It's important to note that these conclusions are based on the provided data, and additional research and analysis may be needed for a more comprehensive understanding of tobacco use patterns in India. Additionally, tobacco use is a complex public health issue influenced by various socio-economic, cultural, and behavioral factors, which further underscores the importance of tailored strategies for effective tobacco control and prevention.